opensourceleg/robots/osl.py

```diff
@@ -0,0 +1,88 @@
+from typing import Any, Dict, Generic, List, NamedTuple, Optional, TypeVar, Union
+
+from opensourceleg.actuators.base import ActuatorBase
+from opensourceleg.actuators.dephy import DephyActuator
+from opensourceleg.logging import LOGGER
+from opensourceleg.logging.exceptions import ActuatorKeyException
+from opensourceleg.robots.base import RobotBase, TActuator, TSensor
+from opensourceleg.sensors.base import IMUBase, LoadcellBase, SensorBase
+
+
+class OpenSourceLeg(RobotBase[TActuator, TSensor]):
+    def start(self):
+        super().start()
+
+    def stop(self):
+        super().stop()
+
+    def update(self):
+        super().update()
+
+    def home(self):
+        for actuator in self.actuators.values():
+            actuator.home()
+
+    def make_encoder_maps(self):
+        pass
+
+    @property
+    def knee(self) -> Union[TActuator, ActuatorBase]:
+        try:
+            return self.actuators["knee"]
+        except KeyError:
+            LOGGER.error(
+                "Knee actuator not found. Please check for `knee` key in the actuators dictionary."
+            )
+            exit(1)
+
+    @property
+    def ankle(self) -> Union[TActuator, ActuatorBase]:
+        try:
+            return self.actuators["ankle"]
+        except KeyError:
+            LOGGER.error(
+                "Ankle actuator not found. Please check for `ankle` key in the actuators dictionary."
+            )
+            exit(1)
+
+    @property
+    def loadcell(self) -> Union[TSensor, LoadcellBase]:
+        try:
+            return self.sensors["loadcell"]
+        except KeyError:
+            LOGGER.error(
+                "Loadcell sensor not found. Please check for `loadcell` key in the sensors dictionary."
+            )
+            exit(1)
+
+    @property
+    def imu(self) -> Union[TSensor, IMUBase]:
```

Comment:
I think base OSL shouldn't have an IMU... for things people get from Humotech, they'll just have the load cell and two actuators.